src/meta/processors/admin/AdminClient.cpp

```diff
@@ -387,19 +394,24 @@ void AdminClient::getResponse(std::vector<HostAddr> hosts,
                     getResponse(std::move(hosts),
                                 index,
                                 std::move(req),
-                                remoteFunc,
+                                std::move(remoteFunc),
                                 retry + 1,
                                 std::move(p),
-                                retryLimit);
+                                retryLimit,
+                                statisResult);
                     return;
                 }
                 p.setValue(Status::Error(folly::stringPrintf("RPC failure in AdminClient: %s",
                                                              t.exception().what().c_str())));
                 return;
             }
-            auto&& result = std::move(t).value().get_result();
+            auto&& result = std::move(t.value().get_result());
             if (result.get_failed_parts().empty()) {
+                // succeeded
                 p.setValue(Status::OK());
+                if (statisResult && t.value().__isset.statis) {
```

Comment:
This is so weird - -. Maybe use template, or at least just add a new function instead of effect others.

Reply:
The `Status `is currently returned. When all `Status `are ok, start statistics
`cpp2::StatisItem.`

What you said makes sense, but in that case, more code changes.
I think the smallest code changed like this. WDYT?

Reply:
At least don't add a `cpp2::StatisItem* statisResult`, you could pass a function to do whatever you like. Other methods in `AdminClient` just pass an empty function. If you just add a pointer, subsequent changes will add more and more pointer.